frontend/src/actions/authActions.ts

```diff
@@ -35,10 +39,11 @@ export const signUserUp = (payload: IUserDetails) => async (dispatch:Dispatch<fe
 
   try {
     const result = await authApiClient.register(payload)
-    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
-    //@ts-ignore
     localStorage.setItem('token', result.token.access)
-    dispatch(setUserAction(result.user))
+    dispatch(setUserAction(result))
+     // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    //@ts-ignore
+    dispatch(push('/welcome'))
```

Comment:
You can remove this directives by doing next actions:
 - Import `CallHistoryMethodAction` from 'connected-react-router' in addition to 'push' that is already imported
 - Extend `Dispatch` type on line 37 from `dispatch:Dispatch<fetchUserAction>` to dispatch:Dispatch<fetchUserAction|CallHistoryMethodAction>